src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
+    Vec3f sp68;
+    s16 sp66;
+    s16 sp64;
+    s16 sp62;
+    f32 temp_f22;
+    f32 temp_f20;
+
+    Math_Vec3f_Copy(&sp68, &this->unk_348[0]);
+    sp66 = (Rand_Next() >> 0x12) + this->actor.shape.rot.y;
+    temp_f20 = Math_CosS(sp66);
+    temp_f22 = Math_SinS(sp66);
+
+    if (this->unk_32C.r > 30) {
+        this->unk_32C.r -= 16;
+        this->unk_32C.g += -0x10;
+    }
+
+    if (this->unk_32C.b < 30) {
+        this->unk_32C.b += 5;
+        this->unk_32C.a += 8;
+        this->unk_330.a += 8;
+    }
+
+    if (this->unk_330.r != 0) {
+        this->unk_330.r -= 15;
+    }
+
+    if (this->unk_330.g != 0) {
+        this->unk_330.g--;
+    }
+
+    sp80.x = temp_f22 * 3.5f * this->unk_334;
+    sp80.y = this->unk_330.r * 0.02f * this->unk_334;
+    sp80.z = temp_f20 * 3.5f * this->unk_334;
+    sp74.x = (Rand_ZeroFloat(0.1f) + 0.15f) * -temp_f22 * this->unk_334;
+    sp74.y = this->unk_334 * 0.3f;
+    sp74.z = (Rand_ZeroFloat(0.1f) + 0.15f) * -temp_f20 * this->unk_334;
+    sp64 = this->unk_334 * 100.0f;
+    sp62 = this->unk_334 * 25.0f;
+    func_800B0EB0(globalCtx, &sp68, &sp80, &sp74, &this->unk_32C, &this->unk_330, sp64, sp62, 0x14);
+
+    sp66 = ((u32)Rand_Next() >> 0x13) + this->actor.shape.rot.y;
+    temp_f20 = Math_CosS(sp66);
+    temp_f22 = Math_SinS(sp66);
+
+    sp68.x -= (temp_f20 * 6.0f) * this->unk_334;
+    sp68.z += (temp_f22 * 6.0f) * this->unk_334;
+
+    sp80.x = -temp_f20 * 3.5f * this->unk_334;
+    sp80.z = temp_f22 * 3.5f * this->unk_334;
+
+    sp74.x = (Rand_ZeroFloat(0.1f) + 0.15f) * temp_f20 * this->unk_334;
+    sp74.z = (Rand_ZeroFloat(0.1f) + 0.15f) * -temp_f22 * this->unk_334;
+    func_800B0EB0(globalCtx, &sp68, &sp80, &sp74, &this->unk_32C, &this->unk_330, sp64, sp62, 0x14);
+
+    sp68.x = this->unk_348[0].x + ((temp_f20 * 6.0f) * this->unk_334);
+    sp68.z = this->unk_348[0].z - (temp_f22 * 6.0f * this->unk_334);
+    sp80.x *= -1.0f;
+    sp80.z = sp80.z * -1.0f;
+    sp74.x = (Rand_ZeroFloat(0.1f) + 0.15f) * -temp_f20 * this->unk_334;
+    sp74.z = (Rand_ZeroFloat(0.1f) + 0.15f) * temp_f22 * this->unk_334;
+    func_800B0EB0(globalCtx, &sp68, &sp80, &sp74, &this->unk_32C, &this->unk_330, sp64, sp62, 0x14);
+}
+
+s32 func_8087721C(EnDodongo* this) {
+    if ((this->actionFunc == func_8087784C) && (this->skelAnime.animCurrentFrame >= 29.0f) &&
+        (this->skelAnime.animCurrentFrame <= 43.0f)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_80877278(EnDodongo* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    Actor* explosive = globalCtx->actorCtx.actorList[ACTORCAT_EXPLOSIVES].first;
+    Vec3f sp44;
+
+    while (explosive != NULL) {
+        if (!explosive->params && (explosive->parent == NULL) && (explosive->update != NULL) &&
+            (explosive != player->unk_388) &&
+            (((explosive->id == ACTOR_EN_BOM) && (((EnBom*)explosive)->timer > 0)) ||
+             ((explosive->id == ACTOR_EN_BOMBF) && (((EnBombf*)explosive)->timer > 0)))) {
```

Comment:
No more filling the heap when eating a bombchu. You could put a note that this is a bugfix for the OoT version.